trucks.js

```diff
@@ -503,9 +505,80 @@ var foodTrucks = [
   }
 ];
 
-// this module should support the following methods:
-// getTrucks() - return all trucks
+// this module should support the following methods: getTrucks() - return all trucks
+var getTrucks = function() {
+  var allTrucks = [];
+  for (var i = 0; i < foodTrucks.length; i++) {
+    allTrucks.push(foodTrucks[i].name );
+  }
+
+  return allTrucks;
+};
+
 // getTruck(name) - return the truck object matching 'name'
+var getTruck = function(name) {
+  var thisTruck =[];
+  for (var i = 0; i < foodTrucks.length; i++) {
+    var nameOfTruck = foodTrucks[i].name.toString();
+    if (name === nameOfTruck) {
+      thisTruck.push( foodTrucks[i]);
+    }
+  }
+
+  return thisTruck;
+};
+
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
+var getFoodTypes = function() {
+  var foodType =[];
+  for (var i = 0; i < foodTrucks.length; i++) {
+    var typeOfFood = foodTrucks[i].type.toString();
+    foodType.push( typeOfFood );
+  }
+
+  return foodType;
```

Comment:
The `type` property is an array of values. You ultimately want to return a list of values that looks like this:

`['BBQ', 'Sweets', 'Hamburgers']`

but instead you're going to end up returning something that looks like this:

`[['BBQ'], ['Sweets'], ['Pies, Sweets']]`

There problem here is two-fold. First, You don't want to return an array of arrays, you just want to return an array of values. Look at the _.flatten method of lodash to see how you can do this. Second, you're going to end up with duplicates. Look at the _.uniq method of lodash to see how you can fix that.